### PR TITLE
New Port: Super Amazing Wagon Adventure

### PR DESCRIPTION
**Pull request by JanTrueno**
## New Port: Super Amazing Wagon Adventure
**Description:** 
Super Amazing Wagon Adventure is a fast-paced 2d shooting game with a random story. You control a group of three migrants traveling west along a wagon trail. The game consists of short scenes that are sequenced together in a semi-random way to tell a crazy adventure story.

**Details:**
|Detail  | info |
|--|--|
| Engine | XNA|
| Architectures | 64bit|
| Aspect ratio | keeps 16:9  |
| Rumble support | Yes |
| Tested versions| [Steam](https://store.steampowered.com/app/250500/Super_Amazing_Wagon_Adventure/)| 
| Controls | native |

**Instructions:**
- Install portfiles with PM autoinstaller.
- Buy and install Super Amazing Wagon Adventure
- Copy all files to  the "wagonadvenure/gamedata" folder.

**Controls:**
|Controls|Action|
|--|--|
|A/B|Shoot|
|D-pad|Move|
|Start|Menu

## Requirements:
**Custom Firmwares:**
- [X] AmberELEC
- [X] ArkOS
- [x] ROCKNIX
- [x] MuOS
- [X] Knulli**
- [X] Crossmix **

**Drivers:** 
- [x] Mali
- [x] Panfrost
- [x] Adreno **

**Resolutions:** 
- [X] 480x320 **
- [x] 640x480
- [X] 720x720**
- [x] Higher resolutions 

## Notes:
- Thanks to [SparseVector](https://store.steampowered.com/search/?developer=sparsevector) for making this game.
- Make sure to have the mono runtime installed :slight_smile:
- Thanks to the FNA Framework devs.

-# ** = Optional requirement

### CODE REVIEW

File: ports/superamazingwagonadventure/Super Amazing Wagon Adventure.sh [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  controlfolder="$XDG_DATA_HOME/PortMaster"
+else
+  controlfolder="/roms/ports/PortMaster"
+fi
+
+# Pm
+source $controlfolder/control.txt
+[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
+get_controls
+source $controlfolder/tasksetter
```
Comment (Cebion) [inline ports/superamazingwagonadventure/Super Amazing Wagon Adventure.sh:19]: Please add a little comment why you're using tasksetter

File: ports/superamazingwagonadventure/Super Amazing Wagon Adventure.sh [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+$ESUDO umount "$monofile" || true
+$ESUDO mount "$monofile" "$monodir"
+
+# Setup savedir linux
+bind_directories "$HOME/.local/share/WagonAdventure" "$GAMEDIR/savedata" 
+
+# Setup path and other environment variables
+export MONO_PATH="$GAMEDIR/dlls"
+export LD_LIBRARY_PATH="$GAMEDIR/libs:$LD_LIBRARY_PATH"
+#export LD_LIBRARY_PATH="$GAMEDIR/libs:/usr/config/emuelec/lib32:/usr/lib32:$LD_LIBRARY_PATH"
```
Comment (Cebion) [inline ports/superamazingwagonadventure/Super Amazing Wagon Adventure.sh:44]: Not needed? Remove

File: ports/superamazingwagonadventure/gameinfo.xml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<gameList>
+  <game>
+    <path>./Super Amazing Wagon Adventure.sh</path>
+    <name>Super Amazing Wagon Adventure</name>
+    <desc>Super Amazing Wagon Adventure is a fast-paced 2d shooting game with a random story. You control a group of three migrants traveling west along a wagon trail. The game consists of short scenes that are sequenced together in a semi-random way to tell a crazy adventure story.</desc>
+    <releasedate>20131017T000000</releasedate>
+    <developer>SparseVector</developer>
+    <genre>Adventure</genre>
+    <image>./superamazingwagonadventure/cover.png</image>
```
Comment (Cebion) [inline ports/superamazingwagonadventure/gameinfo.xml:10]: What's with the long path. Should start with s sure but can be shortened no?
superwagon or whatever.

werenotdoingthesekindoflongnamesbecauseno

**Review by Cebion — CHANGES_REQUESTED** [review]

**Review by kloptops — APPROVED** [review]
Looks good, shortening the directory names wouldn't hurt.

**Review by Cebion — APPROVED** [review]

**Review by cdeletre — APPROVED** [review]

**Pull request merged**